toolz/utils.py

```diff
@@ -6,4 +6,11 @@ def raises(err, lamda):
         return True
 
 
-no_default = '__no__default__'
+@object.__new__
+class no_default(object):
+    def __new__(self):
+        raise TypeError("cannot create 'no_default' instances")
+
+    def __str__(self):
+        return '<object no_default>'
+    __repr__ = __str__
```

Comment:
I'm considering the use of either

    no_default = object()

or 

    no_default = '__no_default__'   # same as before

I guess that the first reprs oddly in help strings.  What was the objection to the second?  Or more generally, are we confident that this is the lowest-tech way to achieve a sentinel value?